projects/services-ogc/src/lib/owc/owc-json.service.ts

```diff
@@ -158,88 +210,138 @@ export class OwcJsonService {
+    if (resource.properties?.opacity) {
       opacity = resource.properties.opacity;
     }
     return opacity;
   }
 
-  getResourceAttribution(resource: IOwsResource): string {
+  /** OWS Extenson IEocOwsResource */
+  getResourceAttribution(resource: IEocOwsResource): string {
     let attribution = '';
-    if (resource.properties.hasOwnProperty('attribution')) {
+    if (resource.properties?.attribution) {
       attribution = resource.properties.attribution;
     } else if (resource.properties.rights) {
       attribution = resource.properties.rights;
     }
     return attribution;
   }
 
-  getResourceShards(resource: IOwsResource): string {
-    if (resource.properties.hasOwnProperty('shards')) {
+  /** OWS Extenson IEocOwsResource */
+  getResourceShards(resource: IEocOwsResource): string {
+    if (resource.properties?.shards) {
       return resource.properties.shards;
     }
   }
 
+  /** OWS Extenson IEocOwsResource */
+  getResourceMinMaxZoom(resource: IEocOwsResource, targetProjection: string = 'EPSG:4326'): { minZoom: number; maxZoom: number; } {
+    const zooms = { minZoom: null, maxZoom: null };
+    if (resource.properties.minZoom) {
+      zooms.minZoom = resource.properties.minZoom;
+    } else if (resource.properties.maxscaledenominator) {  // *Max*ScaleDenom ~ *Min*Zoom
+      zooms.minZoom = this.scaleDenominatorToZoom(resource.properties.maxscaledenominator, targetProjection) || null;
+    }
+    if (resource.properties.maxZoom) {
+      zooms.maxZoom = resource.properties.maxZoom;
+    } else if (resource.properties.minscaledenominator) {  // *Min*ScaleDenom ~ *Max*Zoom
+      zooms.maxZoom = this.scaleDenominatorToZoom(resource.properties.minscaledenominator, targetProjection) || null;
+    }
+    return zooms;
+  }
+
 
   /**
    * e.g.
    * (array)   value: '1984-01-01T00:00:00.000Z/1989-12-31T23:59:59.000Z/PT1S,1990-01-01T00:00:00.000Z/1994-12-31T23:59:59.000Z/PT1S,...'
+   * (array)   value: '1984-01-01T00:00:00.000Z/P1D,P1D/2000-01-01T00:00:00.000Z,...'
    * (array)   value: '2000-01-01T00:00:00.000Z,2001-01-01T00:00:00.000Z,2002-01-01T00:00:00.000Z,...'
    * (single) value: '2016-01-01T00:00:00.000Z/2018-01-01T00:00:00.000Z/P1Y'
    */
-  getTimeValueFromDimensions(input: string | null): string | string[] | ILayerIntervalAndPeriod | ILayerIntervalAndPeriod[] {
-    if (!input) {
-      return null;
-    }
-
-    const splitValues = input.split(',');
-    if (splitValues.length > 0) {
-      const outputs = [];
-      for (const value of splitValues) {
-        const parsed = this.parseSingleTimeOrPeriod(value);
-        outputs.push(parsed);
+  getTimeValueFromDimensions(values: IEocOwsTimeDimension['values'], period?: IEocOwsTimeDimension['display']['period']): ILayerIntervalAndPeriod | Array<string | ILayerIntervalAndPeriod> {
+    if (values === null) {
+      return;
+    } else {
+      const isList = /,/g.test(values);
+      if (isList) {
+        // values: `${string},${string}`
```

Comment:
Here we check for separated steps like:
- 2020-01-08T16:07:00.000Z,2020-01-20T16:07:00.000Z,2020-02-01T16:07:00.000Z
- 1984-01-01T00:00:00.000Z/1989-12-31T23:59:59.000Z/P1Y,1990-01-01T00:00:00.000Z/1994-12-31T23:59:59.000Z/P1Y,1995-01-01T00:00:00.000Z/1999-12-31T23:59:59.000Z/P1Y
- 1984-01-01T00:00:00.000Z/1989-12-31T23:59:59.000Z/P1Y,1990-01-01T00:00:00.000Z,1991-01-01T00:00:00.000Z,1992-01-01T00:00:00.000Z,1993-01-01T00:00:00.000Z/1994-12-31T23:59:59.000Z